### DISCUSSION TITLE
Unable to screenshare in Niri

### DISCUSSION THREAD

**Discussion by ConsularParadi** [YaLTeR/niri#784] [Q&A]
I wasn't aware that niri did not have screensharing built in and just installed xdg-desktop-portal-gnome and rebooted my system but the service fails with 
` Dependency failed for Portal service (GNOME implementation).`
Also I was running the cmd niri when entering through greetd-tuigreet but have changed it to `niri --session`.
Are there any troubleshooting steps that I should go through?

**Comment by YaLTeR** [comment]
Installing xdp-gnome and starting niri through niri-session script should be sufficient. Also make sure pipewire is running before niri

**Reply by ConsularParadi** [reply]
Hey, I have tried adding pipewire to my fish config as suggested by a user using `nohup pipewire > /dev/null` but it still gives out the same error
`xdg-desktop-portal-gnome.service: Job xdg-desktop-portal-gnome.service/start failed with result 'dependency'.`
And screensharing still fails. Earlier I was able to select between which screen I wanted to share and it showed a black screen with my mouse pointer but since I have install xdp-gnome, it fails to even do that.

**Reply by YaLTeR** [reply]
Check systemd logs, why xdp-gnome fails to start. Maybe you didn't install some dependency like xdg-desktop-portal?

**Reply by ConsularParadi** [reply]
This is the error I am getting in the logs.
```
Current graphical user session is inactive.
Dependency failed for Portal service (GNOME implementation).
xdg-desktop-portal-gnome.service: Job xdg-desktop-portal-gnome.service/start failed with result 'dependency'.
```

There was also one log which I have not been able to reproduce which said `cannot open display`. I had already added DISPLAY=:0 in niri config file in environment when setting up xwayland satellite. Also, the xdg-desktop-portal service is already running.

**Reply by YaLTeR** [reply]
Strange. This shouldn't happen if you use the `niri-session` script since it brings up niri via its systemd service, which should pull the graphical session.

**Reply by ConsularParadi** [reply]
Does `niri --session` have the same effect when run via tty ? I might be doing something wrong here because I have not setup any systemd services for niri and it runs through tty using `niri --session` command.

**Reply by YaLTeR** [reply]
It doesn't bring up the systemd services, no

**Reply by ConsularParadi** [reply]
Ok, I will try setting up niri through systemd. Thank you for helping.

**Reply by ConsularParadi** [reply]
Hi, running niri-session using command `tuigreet --cmd niri-session` leads to a tty upon login. Is this the intended result? If so, what is the recommended way of entering niri post that?
Greetd is just a login daemon and not a display manager so are there any other steps for setting up dependencies for niri if using greetd?

**Reply by YaLTeR** [reply]
Not sure how to do it with greetd, but I think someone is successfully using it

**Reply by ConsularParadi** [reply]
Hi @YaLTeR, I searched through [tuigreet](https://github.com/apognu/tuigreet) which is the greeter I am using for greetd. They mention this in their setup guide
```
greetd only accepts environment-less commands to be used to start a session. Therefore, if your desktop environment requires either arguments or environment variables, you will need to create a wrapper script and refer to it in an appropriate desktop file.
```
This is the default desktop entry created for niri by greetd and which is the current file being used while logging in.
```
[Desktop Entry]
Name=Niri
Comment=A scrollable-tiling Wayland compositor
Exec=niri-session
Type=Application
DesktopNames=niri
```
I also saw another example of desktop entry which was taken from [this thread](https://github.com/apognu/tuigreet/issues/155) and looks like this
```
[Desktop Entry]
Exec=/usr/lib/plasma-dbus-run-session-if-needed /usr/bin/startplasma-wayland
TryExec=/usr/bin/startplasma-wayland
DesktopNames=KDE
Name=Plasma (Wayland)
```
I created a wrapper script as follows
`XDG_SESSION_TYPE=wayland dbus-run-session niri-session`
but it still takes me to the terminal instead of opening the window manager

If possible, could you help me figure out the mistake I am making in creating a custom entry for niri or if there is any problem with the original one ?

**Reply by YaLTeR** [reply]
When it takes you to the terminal, does it say anything? Anything in `journalctl --user-unit=niri -b`?

**Reply by ConsularParadi** [reply]
It says `No Entries`.
I'm not sure if this is relevant but in both cases (the original script and the wrapper script) `systemctl --user status niri` also returns `inactive (dead)`.
Also I was using `tuigreet --cmd niri` in greetd config file which I assume first runs `niri-session` upon login as mentioned in the desktop entry and then runs niri command to directly throw me into the GUI

**Reply by YaLTeR** [reply]
Seems like systemd never starts it?

**Reply by ConsularParadi** [reply]
It seems so. I'll hop onto tuigreet's repo to inquire on how to setup niri with greetd.
Just to be clear, when running in absence of a display manager, `niri-session` is a binary that needs to be run to setup all related services and environment variables for the system to function?
And should the GUI be automatically accessible or the `niri` command needs to be run to access that?

**Reply by YaLTeR** [reply]
You can cat the niri-session. It's a shell script that more or less boils down to starting niri.service, which itself runs `niri`. You should be able to run it on the tty. I use GDM with the .desktop file from this repo (same as you showed above).

**Comment by ConsularParadi** [comment]
Hi, I was able to get the screencasting working with `xdp-gnome`. The main issue occurs when logging into niri using `greetd tuigreet`.
What works for me is using the available `.desktop` file for niri and `tuigreet` as command in `/etc/greetd` without `--cmd` suboption.
I am thrown into the tty post login using this config and when I press `ctrl+d`, the niri wm starts with screensharing enabled.

**Reply by YaLTeR** [reply]
Thanks for sharing the solution

**Reply by theodotdot** [reply]
Hey sorry to revive this but I have had the same issue recently with niri and greetd, I have tried your solution but it doesn't seem to work, are you still using niri with tuigreet and managing to have xdp-gnome working correctly?
